src/microbe_stage/systems/EngulfingSystem.cs

```diff
@@ -195,6 +195,8 @@ protected override void Update(float delta, in Entity entity)
 
         var actuallyEngulfing = control.State == MicrobeState.Engulf && cellProperties.MembraneType.CanEngulf;
 
+        cellProperties.CreatedMembrane?.EnableEngulfAnimation(actuallyEngulfing, delta);
```

Comment:
I just added handling for this if you could look over the MicrobeShaderSystem.cs changes. With those changes it fades out nicely.

Reply:
I just realized, should I add a ReadsComponent Attribute if the component is optional?

Reply:
From reading other systems, it appear I should, so I will add that now.

Reply:
Yes, if you added a dependency on a new component on a system, it needs adjusting like that. And probably need to also run the generate threaded run scene to make sure there are no component dependency problems in the system execution order.

Reply:
In case you didn't find it the script can be ran like: `godot . res://src/tools/GenerateThreadedSystems.tscn` it should then in-place update the generated microbe world simulation file.

Reply:
I just updated the thread order using the script.